pkg/signup/signup_service.go

```diff
@@ -240,25 +242,47 @@ func (s *ServiceImpl) GetSignup(userID string) (*Signup, error) {
 }
 
 // GetUserSignup is used to return the actual UserSignup resource instance, rather than the Signup DTO
-func (s *ServiceImpl) GetUserSignup(userID string) (*crtapi.UserSignup, error) {
+func (s *ServiceImpl) GetUserSignup(userID string) (*v1alpha1.UserSignup, error) {
 	// Retrieve UserSignup resource from the host cluster
 	userSignup, err := s.UserSignups.Get(userID)
 	if err != nil {
-		if errors.IsNotFound(err) {
-			return nil, nil
-		}
 		return nil, err
 	}
 
 	return userSignup, nil
 }
 
 // UpdateUserSignup is used to update the provided UserSignup resource, and returning the updated resource
-func (s *ServiceImpl) UpdateUserSignup(userSignup *crtapi.UserSignup) (*crtapi.UserSignup, error) {
+func (s *ServiceImpl) UpdateUserSignup(userSignup *v1alpha1.UserSignup) (*v1alpha1.UserSignup, error) {
 	userSignup, err := s.UserSignups.Update(userSignup)
 	if err != nil {
 		return nil, err
 	}
 
 	return userSignup, nil
 }
+
+// PhoneNumberAlreadyInUse checks if the phone number has been banned. If so, return
+// an internal server error. If not, check if a signup with a different userID
+// exists. If so, return an internal server error. Otherwise, return without error.
+func (s *ServiceImpl) PhoneNumberAlreadyInUse(userID, countryCode, phoneNumber string) error {
```

Comment:
Ths function logic looks good now, however I'm wondering if this protection could be defeated by the user by simply inserting spaces into the phone number.  Perhaps we should "normalize" these values by removing all spaces.  Possibly we should also enforce some form of validation to ensure the user can only enter digits, spaces and brackets `()`, and then to normalize remove everything that's not a digit?  WDYT?

Reply:
this is already addressed..if spaces are given this error is shown:
example input: 
{"country_code":"     ", "phone_number":"          "}
```
{
    "status": "Internal Server Error",
    "code": 500,
    "message": "A 'To' phone number is required.",
    "details": "error while verifying code"
}
```

Reply:
in the case that letters are given, this is the error message: 
{"country_code":"", "phone_number":"abadsdsdfsfsfdsdffd"}
```
{
    "status": "Internal Server Error",
    "code": 500,
    "message": "The 'To' number  is not a valid phone number.",
    "details": "error while verifying code"
}
```

Reply:
there are different variations of this error message depending on what is entered (for instance is the country_code not valid and/or the phone_number is not valid)

Reply:
i have tested this via postman, @sbryzak if you could please also test and make sure everything is fine, that would be a great help. i have tested different variations and nothing comes back as "positive" when invalid data is given. these error messages don't come from us, they come from twilio.

Reply:
Are you saying this is validated by twilio? So, if the user entered a phone number with spaces then it will fail?
Shouldn't we normalize the phone number (delete all spaces) as Shane suggested?

Reply:
yeah, twilio already performs basic phone number validation.

> Twilio's API has some very basic phone number validation logic built in. We try to save you money by rejecting messages to numbers we are certain are incorrect. However, we trust that our users know more about the phone number they are trying to send a message to than we do. Our validation logic will only affect numbers which are unquestionably incorrect. It is your responsibility to provide accurate data. 

[source: https://support.twilio.com/hc/en-us/articles/223181828-Does-Twilio-check-to-see-if-phone-numbers-can-receive-SMS-]

however, we could add our own basic phone number validation to be extra safe - WDYT? The things we don't validate against currently are things like validating if a number from a certain region is correct - do we want to add this?

the UI should block a lot of the basic problematic phone number entries.. currently the code in the UI does not handle this

Reply:
It's not only about validation.
What if the user enter the phone number as:
- `(925) 123-4567`
- `925 123-45-67`
- `925 123 4567`
We will fail, right? So, we need to "normalize" them. Either in UI (then our backend should expect only numbers and fail with anything else) or in the backend. I would do the normalization in our backend. Just delete all not digit symbols from the phone string. So all the examples above will be normalized to the same string: `9251234567`. And it must be done in the controller level. So we use the normalized number everywhere.

Reply:
no, these cases are valid with twilio. i tried each one of them with my own phone number and it works.. i guess as the document states it will only do basic validation, so as long as the number looks correct it will not fail, hence no need to normalize. they already do the normalization and validation on their end..

Reply:
also Michael added some validation to the UI as well using regex so it's covered on that end as well

Reply:
OK. But then if the same user tries the same number, first time as 1234567 and the second time as 123 4567, then we will treat them as two different numbers. But we shouldn't. So we have to normalize them on our backend side.

Reply:
ohhh right, because of the hash, i've updated it as part of: 43237d7c8084c491f3f9539a329e2fe6a7d8cb3f is this okay?